### PR TITLE
[dompurify] Expose dompurify typings on namespace and update types to better reflect version 2.0.1

### PR DESCRIPTION
**Pull request by Exigerr**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/cure53/DOMPurify/issues/367>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

cure53/DOMPurify#367

### CODE REVIEW

**Review by demensky — APPROVED** [review]

**Review by MarsiBarsi — APPROVED** [review]

**Review by waterplea — APPROVED** [review]
5 minutes before I was about to make my own PR ))

**Comment by waterplea** [comment]
Hi @RyanCavanaugh, you've merged last pull request to this package. Would you have time to look at this one please and merge if it's fine? Last one broke some of the typings and this one fixes these issues as well as adds support for newly released DOMPurify 2.0

**Comment by waterplea** [comment]
@bazuzi @FlowCrypt @RyanCavanaugh please could anybody merge this pull request? It's been here for almost a month. Current types for DOMPurify are broken and older version of DOMPurify reports as critical: Cross-site scripting vulnerability by npm. We really need this merged so we can update our dependent TS libraries.

**Comment by waterplea** [comment]
Come on, guys, 30+ days for updating types when current types are broken. I'm sorry to bother everybody again but this needs to be merged. Including everybody who merged PRs to DOMPurify types before since they supposedly have rights to merge it:
@DanielRosenwasser @RyanCavanaugh @sandersn @horiuchi @vvakame @bazuzi @FlowCrypt

File: types/dompurify/index.d.ts [hunk 1]
```diff
@@ -1,87 +1,95 @@
-// Type definitions for DOM Purify
+// Type definitions for DOM Purify 2.0.1
```
Comment (jessetrinity) [inline types/dompurify/index.d.ts]: Should just be 2.0. The patch version here is updated by us and is not meant to match the original package. Please set enable the `dt-header` rule in `tslint.json`.

**Review by jessetrinity — CHANGES_REQUESTED** [review]

**Comment by jessetrinity** [comment]
Does anyone know what is going on with the owners? Ideally when you make changes you should add yourself as an owner, but the assumption is that one of the current owners will approve adding you.

File: types/dompurify/index.d.ts (hunk 1 above)
Comment (Exigerr) [reply types/dompurify/index.d.ts]: Done :)

Should I also commit the `dt-header` flag set to enabled in the `tslint.json` as well?
That will cause a lint failure due to the `Definitions by` containing a non-Github url

**Comment by Exigerr** [comment]
> Does anyone know what is going on with the owners? Ideally when you make changes you should add yourself as an owner, but the assumption is that one of the current owners will approve adding you.

**Comment by waterplea** [comment]
Guys, what happened? @Exigerr, why have you closed this PR?

**Comment by Exigerr** [comment]
> Guys, what happened? @Exigerr, why have you closed this PR?

Apologies. Accidentally hit "close and comment" instead of just comment. Sorry!

**Review by demensky — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Review by jessetrinity — APPROVED** [review]

**Pull request merged**